### PR TITLE
Use the workaround if we can't determine the server's version

### PR DESCRIPTION
**Pull request by bartbroere**
This is related to issue #580 

In some scenarios (a limited-scope API key, or maybe a corporate environment) a user of `eland` might not have enough permissions to determine the host's Elasticsearch version.

In that case, we could default to using the workaround, to be sure.

### CODE REVIEW

**Comment by pquentin** [comment]
buildkite test this please

**Comment by pquentin** [comment]
buildkite test this please

File: eland/field_mappings.py [hunk 1]
```diff
@@ -942,7 +943,19 @@ def _compat_field_caps(client, fields, index=None):
...
+    try:
+        elastic_version = es_version(client)
+    # If we lack sufficient permission to determine the Elasticsearch version,
+    # to be sure we use the workaround for versions smaller than 8.5.0
+    except elasticsearch.AuthorizationException as e:
+        raise RuntimeWarning(
+            "Couldn't determine Elasticsearch host's version. "
+            "Probably missing monitor/main permissions. "
+            "Continuing with the query string work-around. "
+            "Original exception: " + str(e)
```
Comment (pquentin) [inline eland/field_mappings.py:955]:
```suggestion
            "Original exception: " + repr(e)
```

**Review by pquentin — APPROVED** [review]
I'm happy to do this is it unblocks you and the tests pass, but note that there are other places where we look at the ES version in the same way that would break similarly.

**Comment by bartbroere** [comment]
> I'm happy to do this is it unblocks you, but note that there are other places where we look at the ES version in the same way that would break similarly.

Thanks! I noticed too that it was used in some other places. This one is probably in the place someone is most likely to encounter it. You get to this code by instantiating an eland DataFrame I believe.

Of course I'm still trying to find out if we could grant `monitor/main` privileges to all users in our organisation, but for now this would be very helpful.

File: eland/field_mappings.py (hunk 1 above)
Comment (bartbroere) [reply eland/field_mappings.py]: Thanks, this is the better way indeed!

**Comment by droberts195** [comment]
> Of course I'm still trying to find out if we could grant `monitor/main` privileges to all users in our organisation

`cluster:monitor/main` literally just gives access to the `/` endpoint, which I find hard to believe is a security risk to allow for any user who legitimately has access to the cluster to do anything else. We probably ought to make it easier to grant access to this endpoint alone instead of requiring an obscure action name to be granted. But until then I don’t think it would be a security risk to add it manually to some of the other roles you have defined.

**Pull request merged**